### ISSUE TITLE
Speichern bei Verlassen

### ISSUE THREAD

**Issue by lenilsas** [openjverein/jverein#351]
Mein Vorschlag wäre eine Dialog anzuzeigen wenn eine DetailView verlassen wird, in dem gefragt wird, ob gespeichert werden soll. Vor allem bei geschachtelten Views fände ich es sinnvoll, zB. Beim MitgliedDetailView beim erstellen von Zusatzbeiträgen, Wiedervorlagen, Lehrgängen, Kontoauszügen etc.
Was haltet ihr von dem Vorschlag? Und wenn ja überall oder nur bei geschatelten Views?

**Comment by JohannMaierhofer** [comment]
Ich schaue mir aber auch oft einfach nur die Details an ohne sie zu ändern. Da möchte ich eigentlich nicht beim Verlassen immer einen Dialog bekommen wo ich dann klicken muss.
Wahrscheinlich ist es auch schwer herauszufinden ob etwas geändert wurde oder nicht. Dann könnte man nur fragen wenn wirklich etwas geändert wurde. Das müsste dann wohl für jeden View extra implementiert werden.
Die Alternative wäre noch neben Bearbeiten auch einen Menüpunkt Anzeigen zu haben bei dem man in den View im Read-Only Modus kommt. Den kann man dann auch ohne Rückfrage verlassen. Aber der Modus müsste wohl wieder für jeden View extra implementiert werden. Könnte aber so sein wie bei Buchungen die abgeschlossen sind, einfach den Speichern Button disablen. Und dann möchte man wohl auch den Read-only Mode per Button verlassen können.

**Comment by lenilsas** [comment]
Ja, das stimmt, wen man nur anschaut nerven die Dialoge. Ich dachte das sie nur angezeigt werden, wenn was geändert wurde, das ist wahrscheinlich schwer zu ermitteln.
Wie gesagt sehe ich das Problem hauptsächlich bei den Verschachtelten Views. Das sind:

- Splitbuchungen
- Formularfelder
- Mitglieder

Mein Vorschlag wäre die meisten dieser Verschachtelungen in Dialoge zu verschieben, so wie es auch zum Teil schon gemacht ist. So dass bei Mitgliedern jeweils nur ein Dialog für Lehrgänge, Künftige BG, Zusatzbeitrag, Sollbuchung, Wiedervorlage, Arbeitseinsatz kommt und nicht eine neue View geladen wird.
Bei Mitglieder->Mail, Duplizieren, Kontoauszug, Personalbogen kommt ein Dialog "Vorher speichern?"
Bei Splitbuchungen ist es durch das automatische zurückgehen zur Splitübersicht eigentlich schon gut gelöst.
Bei den Formularfeldern könnt man evtl. auch einen Dialog statt der View machen.

**Comment by JohannMaierhofer** [comment]
Bei den Formularfelder habe ich das auch angepasst so dass man bei Speichern gleich zurück kommt.
Die Idee mit Dialogen hatte ich auch schon.

**Comment by lenilsas** [comment]
Ich meine mal irgendwo von Olav gelesen zu habe, dass das hasChangednicht zuverlässig funktioniert. Daher würde ich es nicht verwenden. Ich glaube ich werde es so wie obern beschrieben umsetzen und die Sub-Views in Dialoge umändern. Die Frage mache ich bei den Buttons in der MitgliedDetailView bei welchen die View verlassen wird.
